### PR TITLE
test condition

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Allow testing conditions.

https://user-images.githubusercontent.com/1444314/156706352-9252b1a2-ec70-410b-b06a-fa1bc4cc4e9c.mp4

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by frenck** [comment]
image

Nice!


I do wonder if the color of the button is clear enough though. It feels like it almost provided no feedback on the result (as in it might be confused with the feedback on the click itself, not the click result).

**Comment by matthiasdebaat** [comment]
We can use the same pattern as we have for Configuration validation:
[Valid prototype](https://www.figma.com/proto/mjh4RQMeMqjc3mjmUxRGid/Triggered-%26-Test?page-id=0%3A1&node-id=2%3A49&viewport=241%2C48%2C0.89&scaling=min-zoom&starting-point-node-id=2%3A49&show-proto-sidebar=1)
[Invalid prototype](https://www.figma.com/proto/mjh4RQMeMqjc3mjmUxRGid/Triggered-%26-Test?page-id=0%3A1&node-id=2%3A326&viewport=241%2C48%2C0.89&scaling=min-zoom&starting-point-node-id=2%3A249&show-proto-sidebar=1)

**Comment by balloob** [comment]
The test button is not about if the condition is valid but if it tests true right now (side effect is validating it or else we can't test it, but most things made via UI will be valid)

**Comment by balloob** [comment]
The test button is not about if the condition is valid but if it tests true right now (side effect is validating it or else we can't test it, but most things made via UI will be valid)

**Comment by bdr99** [comment]
Those affected by color blindness might have a hard time distinguishing between those two colors, considering that red-green color blindness is the most common form of color blindness. It might be helpful to add some kind of additional visual feedback, such as a ✅ or ❌ to indicate whether the condition tested true.

**Comment by balloob** [comment]
Added icons to the progress button

https://user-images.githubusercontent.com/1444314/157312265-3c86539a-a399-4856-8a5e-e235635b7e1a.mp4

**Review by zsarnett — APPROVED** [review]

File: src/components/buttons/ha-progress-button.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import "@material/mwc-button";
-import type { Button } from "@material/mwc-button";
+import { mdiAlertOctagram, mdiCheckBold } from "@mdi/js";
```
[the remark this answers is not in the document]
Comment (balloob) [reply src/components/buttons/ha-progress-button.ts:2]: because I felt like X could also indicate bad config.

**Pull request merged**